olp-cpp-sdk-core/tests/client/OlpClientTest.cpp

```diff
@@ -1364,6 +1364,44 @@ TEST_P(OlpClientTest, SlowDownError) {
   EXPECT_EQ(olp::client::ErrorCode::SlowDown, api_error.GetErrorCode());
 }
 
+TEST_P(OlpClientTest, ApiKey) {
```

Comment:
Maybe, it makes sense to add negative tests, e.g. api_key_provider or settings is nullptr to be sure it works properly and no crash occur.
Probably its tested indirectly by other unit tests, but still its nice to have :-)

Reply:
Yes, please add negative tests. If we do not have any test showing that token is taken is api_key_provider is nullptr then please also add one.